### PR TITLE
Fix persisted state when navigating across technologies, while filter is applied

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 90558554

## Summary

Fixes an issue with the sidebar navigator, where it shows no children, even if there are. 
This can happen if you go to another technology, search and then go back.

## Dependencies

NA

## Testing

Steps:
1. Go to any page on a technology
2. In the navigator, click on a link that points to another technology
3. Search for something
4. Go back using the browser Back
5. Assert it does not apply the filter, (its cleared from SessionStorage as well), and it properly renders the children.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by mportiz08** [comment]
@swift-ci test

**Review by mportiz08 — APPROVED** [review]
The code for this is getting pretty hard to follow, but the fix seems to be working well.
